t/lib/server.lua

```diff
@@ -531,6 +531,11 @@ function _M.plugin_proxy_rewrite_resp_header()
     ngx.say(s)
 end
 
+function _M.server_upstream_host()
```

Comment:
There is no need to add a new backend. `/uri` already contains the host: https://github.com/apache/apisix/blob/530fb4d62c990ec23026f43f70b878e157d441bc/t/node/upstream.t#L510